include/oneapi/dpl/iterator

```diff
@@ -56,6 +56,10 @@ using ::std::rbegin;
 using ::std::rend;
 using ::std::reverse_iterator;
 using ::std::size;
+
+using oneapi::dpl::__internal::is_device_accessible_content_iterator;
```

Comment:
This `using` instructions looks strange in this file because (from my point of view) it's some public header file and it should contains only public things... Isn't it?

Reply:
This is making `is_device_accessible_content_iterator_v` publicly available intentionally.  However, perhaps it should be defined directly in a public namespace though... I'm not sure.  

The closest comparison I've seen thus far is `is_execution_policy[_v]`, which we do use internally from its public namespace(s), `oneapi::dpl` and `oneapi::dpl::execution`.  

So, maybe we should directly define it in a non-uglified namespace.

Reply:
I've adjusted this to be defined directly in `oneapi::dpl`